contracts/ccip/GovernanceCCIPRelay.sol

```diff
@@ -32,33 +31,112 @@ contract GovernanceCCIPRelay is IGovernanceCCIPRelay {
   /// @dev Constructor to initialize the GovernanceRelay contract.
   /// @param _timelock The address of the Timelock contract.
   /// @param _router The address of the CCIP router contract.
-  /// @param _destinationChainSelector The chain selector for the destination chain.
-  /// @param _receiver The address of the receiver contract on the destination chain.
+  /// @param _destinationChainSelectors Array of chain selectors for the destination chains.
+  /// @param _destinationReceivers Array addresses of the receiver contracts on the destination chains.
   constructor(
     address _timelock,
     address _router,
-    uint64 _destinationChainSelector,
-    address _receiver
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
   ) {
     ccipRouter = IRouterClient(_router);
     timelock = _timelock;
-    destinationChainSelector = _destinationChainSelector;
-    destinationReceiver = _receiver;
+    _addDestinationChains(_destinationChainSelectors, _destinationReceivers);
   }
 
   /// @inheritdoc IGovernanceCCIPRelay
-  function setDestinationReceiver(address _receiver) external onlyTimeLock {
-    emit DestinationReceiverUpdated(destinationReceiver, _receiver);
-    destinationReceiver = _receiver;
+  function addDestinationChains(
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
+  ) external onlyTimeLock {
+    _addDestinationChains(_destinationChainSelectors, _destinationReceivers);
+  }
+
+  function _addDestinationChains(
+    uint64[] memory _destinationChainSelectors,
+    address[] memory _destinationReceivers
+  ) private {
+    uint256 receiversLength = _destinationReceivers.length;
+    require(
+      _destinationChainSelectors.length == receiversLength,
+      ArrayLengthMismatch()
+    );
+
+    for (uint256 i = 0; i < receiversLength; ) {
+      uint64 _destinationChainSelector = _destinationChainSelectors[i];
+      address _destinationReceiver = _destinationReceivers[i];
+
+      require(
+        ccipRouter.isChainSupported(_destinationChainSelector),
```

Comment:
@voith I think that you might be interested into replicating this check **also** in the `relayMessage` function because as far as I can see from the `Router.applyRampUpdates`, Chainlink could remove the support from a `_destinationChainSelector` at any time.

It's not a **big** problem because `relayMessage` would anyway revert when any of the `Router` functions that use the chain will be invoked, but at least you would handle it with you own logic/custom error.